### PR TITLE
Fix Postgres - extend cluster can cause data loss

### PR DESCRIPTION
**Pull request by plirglo**

### CODE REVIEW

**Comment by przemyslavic** [comment]
It fails just when deploying 1 PostgreSQL node 😢 
```
2021-09-22T14:35:46.8050618Z 14:35:46 INFO cli.engine.ansible.AnsibleCommand - TASK [postgresql : Create lock file to identify one node installation] *********
2021-09-22T14:35:47.5238457Z 14:35:47 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ec2-15-237-143-233.eu-west-3.compute.amazonaws.com]: FAILED! => {"changed": false, "msg": "file (/var/lock/postgres-one-node-cluster.lock) is absent, cannot continue", "path": "/var/lock/postgres-one-node-cluster.lock", "state": "absent"}
```
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-state

`If file, with no other options, returns the current state of path.`

I believe you need to add `state: touch`.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/main.yml [hunk 1]
```diff
@@ -18,6 +18,25 @@
           | map(attribute='name')
           | list }}
 
+- name: Check if postgres-one-node-cluster.lock exists
```
Comment (atsikham) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/main.yml]: In other places we use .flag file extension for such purposes. I suggest do the same here.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/main.yml [hunk 2]
```diff
@@ -18,6 +18,25 @@
...
+  when:  groups['postgresql'] | length > 1
+  stat:
+    path: /var/lock/postgres-one-node-cluster.lock
+  register: postgres_one_node_lock
+
+- name: Set variable _pg_primary_node
+  when:  groups['postgresql'] | length > 1 and postgres_one_node_lock.stat.exists
```
Comment (atsikham) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/main.yml:28]:
```suggestion
  when:
    - groups['postgresql'] | length > 1
    - postgres_one_node_lock.stat.exists
```

**Review by atsikham — CHANGES_REQUESTED** [review]

**Comment by przemyslavic** [comment]
There are still issues. Successfully deployed 1 PostgreSQL node but when scaling up to 2 nodes it fails when new node goes to first place in inventory.

```
2021-09-23T19:52:17.4122520Z 19:52:17 INFO cli.engine.ansible.AnsibleCommand - TASK [postgresql : Extensions | repmgr | Register node with repmgr] ************
2021-09-23T19:52:17.4465956Z 19:52:17 INFO cli.engine.ansible.AnsibleCommand - skipping: [ci-scdevazurceflannel-postgresql-vm-0]
2021-09-23T19:52:17.8868480Z 19:52:17 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-scdevazurceflannel-postgresql-vm-1]: FAILED! => {"changed": true, "cmd": ["/usr/pgsql-13/bin/repmgr", "standby", "register", "--force", "--upstream-conninfo=host=10.1.6.5,\n      user=epi_repmgr,\n      dbname=epi_repmgr,\n      connect_timeout=2", "--upstream-node-id", "1"], "delta": "0:00:00.027354", "end": "2021-09-23 19:52:17.827834", "msg": "non-zero return code", "rc": 1, "start": "2021-09-23 19:52:17.800480", "stderr": "INFO: connecting to local node \"ci-scdevazurceflannel-postgresql-vm-1\" (ID: 1)\nINFO: connecting to primary database\nERROR: provided node ID for --upstream-node-id (1) is the same as the configured local node ID (1)", "stderr_lines": ["INFO: connecting to local node \"ci-scdevazurceflannel-postgresql-vm-1\" (ID: 1)", "INFO: connecting to primary database", "ERROR: provided node ID for --upstream-node-id (1) is the same as the configured local node ID (1)"], "stdout": "", "stdout_lines": []}
```

Before scaling up:
```
[postgresql]
ci-scdevazurceflannel-postgresql-vm-0 ansible_host=x.x.x.239
```
After
```
[postgresql]
ci-scdevazurceflannel-postgresql-vm-1 ansible_host=x.x.x.82
ci-scdevazurceflannel-postgresql-vm-0 ansible_host=x.x.x.239
```


If the new node goes in second place, that's fine:
Before
```
[postgresql]
ci-scdevazurrhflannel-postgresql-vm-0 ansible_host=x.x.x.72
```
After
```
[postgresql]
ci-scdevazurrhflannel-postgresql-vm-0 ansible_host=x.x.x.72
ci-scdevazurrhflannel-postgresql-vm-1 ansible_host=x.x.x.166

```

**Pull request closed without merging**